scalding-spark/src/main/scala/com/twitter/scalding/spark_backend/SparkBackend.scala

```diff
@@ -11,6 +11,38 @@ import scala.collection.mutable.{ Map => MMap, ArrayBuffer }
 
 object SparkPlanner {
   import SparkMode.SparkConfigMethods
+  sealed trait PartitionComputer {
+    def apply(currentNumPartitions: Int): Int
+  }
+
+  final case object IdentityPartitionComputer extends PartitionComputer {
+    def apply(currentNumPartitions: Int): Int = currentNumPartitions
+  }
+  final case class ConfigPartitionComputer(config: Config, scaldingReducers: Option[Int]) extends PartitionComputer {
+    def apply(currentNumPartitions: Int): Int = {
+      val maxPartitions = config.getMaxPartitionCount
+      val getReducerScaling = config.getReducerScaling.getOrElse(1.0D)
```

Comment:
This could do with some comments -- i'm not entirely sure how much sense it'll make here. Since we can hit one of these guards arbitrarily it could have little effect or lots depending a lot on the graph?

Reply:
yeah, the idea is that if you have a graph already that has explicit numbers of reducers set, you can optionally scale this since hadoop vs spark may behave differently wrt max number of tasks...

Comment is a good point.